common/asyncworkflow/queue/consumer/default_consumer.go

```diff
@@ -270,26 +270,30 @@ func getYARPCOptions(header *shared.Header) []yarpc.CallOption {
 	return opts
 }
 
-func decodeStartWorkflowRequest(payload []byte, encoding string) (*types.StartWorkflowExecutionRequest, error) {
-	if encoding != string(common.EncodingTypeJSON) {
+func (c *DefaultConsumer) decodeStartWorkflowRequest(payload []byte, encoding string) (*types.StartWorkflowExecutionRequest, error) {
+	if encoding != string(common.EncodingTypeThriftRW) {
 		return nil, &UnsupportedEncoding{EncodingType: encoding}
 	}
 
-	var startRequest types.StartWorkflowExecutionAsyncRequest
-	if err := json.Unmarshal(payload, &startRequest); err != nil {
+	var thriftObj shared.StartWorkflowExecutionAsyncRequest
+	if err := c.msgDecoder.Decode(payload, &thriftObj); err != nil {
```

Comment:
why pass encoding as a string if the decoder doesn't even look at it?
seems like this should be either "ask the decoder what the type is" or "just always do thrift, no need to accept it as an argument"